client/src/components/NewLocationReview.jsx

```diff
@@ -0,0 +1,862 @@
+        rating: "",
+    });
+       
+    const handlePlaceSelect = () => {
+        const place = autocomplete.getPlace();
+        if (place.address_components) {
+            // Extract city and state from the address components
+            const cityComponent = place.address_components.find(component => component.types.includes('locality'));
+            const stateComponent = place.address_components.find(component => component.types.includes('administrative_area_level_1'));
+            const city = cityComponent ? cityComponent.long_name : '';
+            const state = stateComponent ? stateComponent.short_name : '';
+
+            if (city && state) {
+                navigate(`/listings/${city}/${state}`);
+            }
+        } else {
+            alert("No details available for input: '" + place.name + "'");
+        }
+    };
+
+    const handleChange = (e) => {
+        const { name, value } = e.target;
+        setFormState({
+            ...formState,
+            [name]: value,
+        });
+    };
+    
+    const handleCheck =() => {
+        setSelectedRadio(value);
+    };
+
+    const handleSubmit = async (e) => {
+        e.preventDefault();
+        if (!isLoggedIn) {
+            // Display the warning alert
+            return;
+        }
+        try {
+            const { data } = await loginUser({
+                variables: { ...formState },
+            });
+            Auth.login(data.loginUser.token);
+            setIsLoggedIn(true);
+        } catch (e) {
+            console.error(e);
+        }
+
+        setFormState({
+            location: "",
+            review: "",
+            rating: "",
+        });
+    };
+
+    const handleCancel = () => {
+        onClose();
+      };
+
+
+    return (
+			<Modal isOpen={isOpen} onClose={onClose}>
+				<ModalOverlay />
+				<ModalContent>
+					<ModalHeader textAlign="left" borderBottomWidth="1px">
+						<Box fontSize="lg">Address</Box>
+						<ModalCloseButton
+							onClick={onClose}
+							variant="unstyled"
+							fontSize="lg"
+							position="absolute"
+							right="1rem"
+							top="0.5rem"
+						>
+							X
+						</ModalCloseButton>
+					</ModalHeader>
+					<Box textAlign="center" mt="4">
+						<Input ref={inputRef} placeholder="Enter address" />
+						<Button mt="4" colorScheme="blue" onClick={handlePlaceSelect}>
+							Search
+						</Button>
+					</Box>
+					<ModalBody>
+						<FormControl>
+							<FormLabel fontSize="lg">Rating</FormLabel>
+							<Accordion allowToggle>
+								<AccordionItem>
+									<h2>
+										<AccordionButton>
+											<Box as="span" flex="1" textAlign="left">
+												<Text as="b">Landlord Rating</Text>
+											</Box>
+											<AccordionIcon />
+										</AccordionButton>
+									</h2>
+									<AccordionPanel pb={4}>
+										<Text color="green" as="em" fontSize="md">
+											Responsiveness
+										</Text>
+										<RadioGroup colorScheme="green" onChange={handleCheck}>
```

Comment:
instead of hard coding these in, we can do a loop over that category, to get all the field descriptions.. that way if we change the text later, we don't have to update in two places.